index.js

```diff
@@ -98,7 +106,8 @@ module.exports = fp(function (fastify, options, next) {
       }
 
       const msg = Buffer.allocUnsafe(cipher.length - sodium.crypto_secretbox_MACBYTES)
-      if (!sodium.crypto_secretbox_open_easy(msg, cipher, nonce, key)) {
+
+      if (!key.some(k => sodium.crypto_secretbox_open_easy(msg, cipher, nonce, k))) {
```

Comment:
```suggestion
      if (!key.some(k => sodium.crypto_secretbox_open_easy(msg, cipher, nonce, typeof key === 'function' ? key() : key))) {
```

Reply:
Not sure why this is an issue? It is defined in a config and up to the person using this library to be sure of what they are doing.

Reply:
It's not an issue, but a misunderstanding of what I thought you were trying to accomplish.  I am trying to implement a feature to support key rotation.  I do not have a use case for a key as a function by itself and think it's a separate issue.

Reply:
I have a feeling this might get expensive if the `key` array is long. From what I can read, it always encode with the first key in the array and can decode with all of them, right?

Reply:
Yes it would put the burden of managing the key array length on the application. Would you want to limit the length of the key array? Or show a big warning if there’s more than 2 or 3?
My typical use case is to only have 2 at any given time with only a few actually needing to use the second one. I’m open to ideas about how to make it more performant as well

Reply:
It's ok, let's just document this issue.